### PR TITLE
Fix a crash on Wayland

### PR DESCRIPTION
**Pull request by zzag**
X11IdleDetector should be created only if it's X11, otherwise Charm
will crash in X11IdleDetector::checkIdleness().

On Wayland, unfortunately, there is no any standardized way to get
notified when the user is idle. KWin has a protocol for that purpose
but it's proprietary.

### CODE REVIEW

**Comment by caybro** [comment]
Still makes charm crash under wayland for me :/

**Comment by bam80** [comment]
Interestingly I have that Idle check tick disabled and still get crash.

**Comment by zzag** [comment]
> Still makes charm crash under wayland for me :/

Do you have a backtrace?

**Comment by bam80** [comment]
According to recent Readme update this project is no longer supported, thus no hope for the MR merge.
What I would suggest is to publish the build on Flathub, if we are still going to use it.
Any thoughts?

**Comment by bam80** [comment]
Still crashes for me too, running with `QT_QPA_PLATFORM=xcb` in Wayland session. Idle check tick is disabled:
```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000000000477409 in X11IdleDetector::checkIdleness (this=0x222b960) at /var/home/bam/timetracker~/Charm/Charm/Idle/X11IdleDetector.cpp:61
61	    const int idleSecs = info->ms_since_user_input / 1000;

File: Charm/Idle/IdleDetector.cpp [hunk 1]
```diff
@@ -54,9 +57,11 @@ IdleDetector *IdleDetector::createIdleDetector(QObject *parent)
 #endif
 
 #ifdef CHARM_IDLE_DETECTION_AVAILABLE
-    X11IdleDetector *detector = new X11IdleDetector(parent);
-    detector->setAvailable(detector->idleCheckPossible());
-    return detector;
+    if (QX11Info::isPlatformX11()) {
```
Comment (frankosterfeld) [inline Charm/Idle/IdleDetector.cpp:60]: At least on Fedora 33, this returns true also for Wayland, thus the code is still executed and crashes.

Comment (dfaure-kdab) [reply Charm/Idle/IdleDetector.cpp:60]: KWindowSystem uses

    if (QGuiApplication::platformName() == QLatin1String("xcb"))

do detect "real X11 without wayland"

Comment (zzag) [reply Charm/Idle/IdleDetector.cpp:60]: So does QX11Info::isPlatformX11(). https://code.woboq.org/qt5/qtx11extras/src/x11extras/qx11info_x11.cpp.html#_ZN8QX11Info13isPlatformX11Ev

Either way, it seems like more proper way to fix this crash will be to check if the Screen Saver extension is available.

Comment (frankosterfeld) [reply Charm/Idle/IdleDetector.cpp:60]: I've pushed a fix now, doing exactly that (checking for the extension)

Comment (zzag) [reply Charm/Idle/IdleDetector.cpp:60]: Great, thanks! I guess this MR can be closed now. :)

Comment (bam80) [reply Charm/Idle/IdleDetector.cpp:60]: Glad it's some sort of "alive" still. Thanks.

**Pull request closed without merging**